DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/CourseSetupController.cs

```diff
@@ -94,5 +113,363 @@ public IActionResult AllCourseStatistics()
+            return bulkSelectResult ?? View("AddNewCentreCourse/SetSectionContent", model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<AddNewCentreCourseData>))]
+        [HttpGet]
+        [Route("AddCourse/Summary")]
+        public IActionResult Summary()
+        {
+            var data = TempData.Peek<AddNewCentreCourseData>();
+
+            var model = new SummaryViewModel(data!);
+
+            return View("AddNewCentreCourse/Summary", model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<AddNewCentreCourseData>))]
+        [HttpPost]
+        [Route("AddCourse/Summary")]
+        public IActionResult? Summary(SummaryViewModel model)
+        {
+            var data = TempData.Peek<AddNewCentreCourseData>()!;
+
+            using var transaction = new TransactionScope();
+
+            var centreId = User.GetCentreId();
+            var customisationId = courseService.CreateNewCentreCourse(
+                centreId,
+                data!.Application!.ApplicationId,
+                data.SetCourseDetailsModel!.CustomisationName ?? string.Empty,
+                data.SetCourseDetailsModel.Password,
+                data.SetCourseOptionsModel!.AllowSelfEnrolment,
+                int.Parse(data.SetCourseDetailsModel.TutCompletionThreshold!),
+                data.SetCourseDetailsModel.PostLearningAssessment,
+                int.Parse(data.SetCourseDetailsModel.DiagCompletionThreshold!),
+                data.SetCourseOptionsModel.DiagnosticObjectiveSelection,
+                data.SetCourseOptionsModel.HideInLearningPortal,
+                data.SetCourseDetailsModel.NotificationEmails
+            );
+
+            var tutorials = data.GetTutorialsFromSections()
+                .Select(
+                    tm => new Tutorial(
+                        tm.TutorialId,
+                        tm.TutorialName,
+                        tm.LearningEnabled,
+                        tm.DiagnosticEnabled
+                    )
+                );
+            if (data.SetSectionContentModels != null)
+            {
+                tutorialService.UpdateTutorialsStatuses(tutorials, customisationId);
+            }
+            transaction.Complete();
+
+            TempData.Clear();
+            TempData.Add("customisationId", customisationId);
+            TempData.Add("applicationName", data.Application!.ApplicationName);
+            TempData.Add("customisationName", data.SetCourseDetailsModel!.CustomisationName);
+
+            return RedirectToAction("Confirmation");
+        }
+
+        [HttpGet]
+        [Route("AddCourse/Confirmation")]
+        public IActionResult Confirmation()
+        {
+            var customisationId = (int)TempData.Peek("customisationId");
+            var applicationName = (string)TempData.Peek("applicationName");
+            var customisationName = (string)TempData.Peek("customisationName");
+            TempData.Clear();

```

Comment:
Although it might be good practice to clear temp data, in this context given we have come from a method that has already called `Clear()` I'm not sure we need it again.
Just the act of calling `(int)TempData["customisationId"]` for example instead of `(int)TempData.Peek("customisationId")` would mark this item for deletion from `TempData`.